### PR TITLE
Pre-handle improvements

### PR DESCRIPTION
**Pull request by netopyr**
Updated the pre-handle workflow to focus on signature verification only. Did an initial attempt to integrate signature verification.

### CODE REVIEW

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadata.java [hunk 1]
```diff
@@ -37,41 +39,44 @@
...
         @Nullable TransactionBody txnBody,
         @Nullable AccountID payer,
+        @Nullable SignatureMap signatureMap,
         @NonNull ResponseCodeEnum status,
         @Nullable HederaKey payerKey,
-        @NonNull List<HederaKey> requiredNonPayerKeys,
-        @Nullable Object handlerMetadata,
-        @NonNull List<ReadKeys> readKeys) {
+        @Nullable TransactionSignature payerSignature,
+        @NonNull Map<HederaKey, TransactionSignature> otherSignatures,
```
Comment (tinker-michaelj) [inline hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadata.java:53]: Are the `HederaKey`'s in this map implicitly restricted to cryptographic keys? (I.e., either Ed25519 or secp256k1 public keys; but not key lists, threshold keys, etc.)

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by nathanklick — DISMISSED** [review]
Approved with respect to the Gradle changes only.

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadata.java (hunk 1 above)
Comment (netopyr) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/meta/TransactionMetadata.java:53]: These are the `requiredNonPayerKeys` set by the pre-handle methods. In the handle workflow, I need a simple way to find the signature data that belongs to a specific key.

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by nathanklick — APPROVED** [review]

**Review by mhess-swl — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @netopyr

**Pull request merged**